src/libcore/str/mod.rs

```diff
@@ -292,6 +292,13 @@ impl<'a> DoubleEndedIterator for Chars<'a> {
     }
 }
 
+impl<'a> Chars<'a> {
+    #[unstable(feature = "iter_to_slice", issue = "27775")]
+    pub fn as_str(&self) -> &str {
```

Comment:
I think this should be `-> &'a str`, so that you can continue modifying the iterator.